packages/flutter_tools/lib/src/build_runner/devfs_web.dart

```diff
@@ -316,30 +317,28 @@ class WebAssetServer implements AssetReader {
   @visibleForTesting
   Uint8List getMetadata(String path) => _metadataFiles[path];
 
+  @visibleForTesting
+  /// The base path to serve from.
+  ///
+  /// It should have no leading or trailing slashes.
+  String basePath = '';
```

Comment:
Right now, the only way to use this is by modifying the source code. It would be nice if we could support this via a cli flag or env variable. Thoughts?

Reply:
This would also need to be specified in the application's index.html right? We could consider parsing it out of that file.

Alternatively a cli flag would be the most approachable option, see the existing `--web-` values that are passed through `DebuggingOptions`

Reply:
you could also configure it through the pubspec.yaml, but that feels redundant with the index.html

Reply:
Correct. This needs be added to `index.html` like this:
```html
<base href="/base/path/">
```
I think we should also change the html template to include the `<base>` tag commented out with an explanation:
```html
<!--
  Uncomment the following line if you are serving your web app in a path other
  than the root.

  Fore more details:
  * https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base
-->
<!-- <base href="/foo/"> -->
```

I like the idea of parsing it automatically from the html file. But when do we parse it?

If we do it at server startup, the user could change their index.html file while the server is running. They have to kill and start their server again (not ideal).

Should we parse it every time we receive a request? I think this might work but will lead to slower responses.

Maybe do it at build time and re-parse on hot restart? This is different from today's behavior where changes in `index.html` don't require a hot restart.

Reply:
I think leaving this commented out in the template is a great idea to start with.

parsing the base url out only at server startup is a reasonable compromise? I can't imagine that changing the value is going to be a common edit case, but if it is we can revisit this in the future.

Otherwise, a hot restart would be the best place to do so - that is somewhat consistent with the rest of the development cycle.

Reply:
Do you mind if I do the template changes and the html parsing in a separate PR? I want to keep the PRs small and easy to review.

Reply:
SGTM